### ISSUE TITLE
Clear instructions for Updating App Images 

### ISSUE THREAD

**Issue by tanaybhomia** [zen-browser/desktop#2518]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

I am trying to update my zen browser using the zsync file ? But I am not able to do so. Can I have some instructions on how can I achieve this.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.14

### What platform are you seeing the problem on?

Linux

**Comment by tanaybhomia** [comment]
![image](https://github.com/user-attachments/assets/1fe1cd4c-3ba5-4e82-a82e-74df42fb0829)
this is the output but the app image is not updated.

**Comment by salamijason** [comment]
Weird, I've seen quite a couple of Appimage issues. Have you tried uninstalling and reinstalling? Refer to https://docs.zen-browser.app/guides/manage-profiles to check how you can keep your data.

**Comment by Vitalii4as** [comment]
I had the same problem and somehow fixed it. I was playing with `appimage.sh` and accidentally removed this line:
```sh
update_output=$(zsync -i ~/.local/share/AppImage/$file_base.AppImage -o ~/.local/share/AppImage/$file_base.AppImage "$zsync_file" 2>&1)
```

So this peace of code was executed in the end:
```sh
            echo "Updating Zen Browser..."
            download_until_success "$appimage_url" ~/.local/share/AppImage/$file_base.AppImage "update"
            process_appimage ~/.local/share/AppImage/$file_base.AppImage $file_base
            log_info "(｡♥‿♥｡) Zen Browser updated to the latest!"
```

And it updated the browser. Seems like the problem is in that zsync command. It probably crashes but the error is not shown because it is written in the variable.

If you try to do what I did please backup your data because I'm not sure what problems can appear)

**Comment by Vitalii4as** [comment]
Well, after restart of the system the zen-browser disappeared from the toolbar in my Ubuntu and from the search. It is either a consequence of my actions, or [this issue](https://github.com/zen-browser/desktop/issues/2313)

**Comment by tanaybhomia** [comment]
doesn't the beta build of the app has a button inside the settings which automatically updates the app and prompts to restart ?

**Comment by Vitalii4as** [comment]
@tanaybhomia I don't know about such a button. Anyway, don't think it works with AppImage.

**Comment by tanaybhomia** [comment]
Yes their is, I am using the AppImage version of the beta build as the flatpak version was going to be release 2 3 days later the official beta released.
Go to settings and then search Update
![image](https://github.com/user-attachments/assets/76b5d1ce-eb3e-4d4c-b773-307c3fc3f432)
Just check this

**Comment by salamijason** [comment]
has op tested the appimage again and tried updating using the button in about:preferences?

**Comment by aymennasri** [comment]
> Yes their is, I am using the AppImage version of the beta build as the flatpak version was going to be release 2 3 days later the official beta released.
> Go to settings and then search Update
> ![image](https://github.com/user-attachments/assets/76b5d1ce-eb3e-4d4c-b773-307c3fc3f432)
> Just check this

The official release has this option locked by the devs ig, so there is no search for updates.

**Comment by salamijason** [comment]
strange...just the usual appimage quirkiness ig. can't really give much of a solution here mb. have you tried the other methods of installation?

**Comment by aymennasri** [comment]
I had the flatpak version, but since updating to Linux Mint 22.1 i opted for the AppImage version due to the annoying and constant flatpak dependencies updates.

Edit: Alright so downloading the new AppImage file and replacing the old one with it, seems to work fine.
No idea if this has any repercussions but for now it's good.

2nd edit: Running the script of updates after replacing the AppImage file, results in an error of launch.

**Comment by salamijason** [comment]
perhaps there's something wrong with the script. thanks for the heads up

**Comment by Samueru-sama** [comment]
AppImage integration and updates are handled by [3rd party tools](https://github.com/AppImageCommunity/awesome-appimage?tab=readme-ov-file#desktop-integration) ideally using [appimageupdate](https://github.com/AppImageCommunity/AppImageUpdate) which why the AppImage releases have those `.zsync` files next to them. 

TLDR is use [AM](https://github.com/ivan-hc/AM), [appimaged](https://github.com/probonopd/go-appimage/blob/master/src/appimaged/README.md) or [soar](https://github.com/pkgforge/soar).

[Gearlever](https://github.com/mijorus/gearlever) also works but it does not make use of the .zsync file to update and instead you have to give it the url to keep track of updates. 

[AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher) also works (and uses zsync), but you have to use the alpha releases because the stable release is incompatible with the static appimage runtime, which is used by this appimage since it removes the common dependency to libfuse2.

----------------------------------------------------------------------

To the maintainers of Zen, I would just remove the install script and just tell people to use the above solutions, it does not make correct use of .zsync file and it also has some issues that haven't been fixed like some hardcoded locations.

**Comment by salamijason** [comment]
Thank you for the suggestion, I'll send it off to Cheff. I personally have tried the gearlever method and it worked fine as well.

**Comment by ccaneke** [comment]
Is the only difference between the tar ball and App Image installation scripts that the App image installation script provides an update feature via Help > About Zen? I'm not sure what the selling point of each one is.

Edit:

It seems the App image installation script only provides an update feature via interactive cli.

**Comment by Samueru-sama** [comment]
> Is the only difference between the tar ball and App Image installation scripts that the App image installation script provides an update feature via Help > About Zen? I'm not sure what the selling point of each one is.

The AppImage and the tarball are very similar, in fact the appimage is just the tarball build put into an AppImage, but there are some differences: 

* The AppImage runtime has the portable `.config` and `.home` mode, meaning that is you make a directory next to the AppImage with the name of the AppImage + `.home`, the AppImage runtime will automatically set that location as `$HOME` for the application. This makes it very easy to have the application alongside with its conig files, and also several different versions of the same application with unique settings each, I make use of this for my work web browser.

* The AppImage sits compressed on the disk, meaning it takes considerably less disk space than the tarball, this comes at the downside of a higher launch time but the new runtime being used means this delay only happens in the very first launch, the following launches re as fast as if it was the tarball/native distro package. 

* The other difference is the way they update, the tarball is able to self update thru the GUI, the AppImage can't, instead with the AppImage you have to use some of the tools mentioned [here](https://github.com/zen-browser/desktop/issues/2518#issuecomment-2702003566), I personally prefer this method, because it means I don't need open the application to update it, which becomes an issue when you have several.


Eventually in the future I might start adding some more dependencies to the AppImage, meaning they are no longer similar, the tarball is made for a common linux system that has glibc + ffmpeg + gtk3. gtk3 is likely to be less common in the future so this can be fixed by bundling the gtk3 lib, which is easy with the appimage since in the `AppRun` script we can set the needed env variable to bundle gtk, but not so much with the tarball, it would have to have its own wrapper script as well.
